### DISCUSSION TITLE
disabling steps in goreleaser

### DISCUSSION THREAD

**Discussion by d-rk** [goreleaser/goreleaser#1733] [Ideas, Issue Triage, and General discussions]
Hi,

I'm using goreleaser to publish 
GitHub release, snap, homebrew, multiple docker images.

Occasionally, something fails or has to be redone, while other steps went through successful and don't need to be run.

E.g. I recently changed the snap part leading to a manual review for my snap (which was declined) so I had to do only the snap publishing again.

For this I need to disable all other steps which can be very tedious.
- the logic is sometimes inverted 'release.disable' vs 'snapcrafts.publish'
- the flags for disabling all have different names: 'disable', 'publish', 'skipUpload', skipPush'
- If there are multiple items of the same kind e.g. dockers, they all have to be disabled separately.

There currently is no possibility to disable these step via env variable right?

How do you think about aligning all these fields?
- name them all 'enabled'
- allow setting 'enabled' false for all dockers e.g.
- allow setting them via env variables

**Comment by caarlos0** [comment]
I agree that this is confusing... we should definitely improve it. The `disable` vs `publish` difference is that `disable` usually disables the whole thing, as `publish: false` just don't publish its results (e.g. it will build the docker images but not push it them).

There is no way to do so via environment variables as well, and honestly not sure it will be... I think its more likely we allow it via `--disable=foo` and `--dont-publish=foo` (or something like that)... but first step is the config improvements.

**Comment by d-rk** [comment]
I think switching something off should have the same boolean value. So I would suggest to:
-  add either `enable` and `publish` to all publishing options. Where `enable=false` means the part is skipped completely and for `publish=false` only local building is done. For some of the publishing options `publish=false` will have the same effect like `enabled=false` since there is nothing done locally, but it could be kept for consistency.
- or add `disable` and `skip_publish` with the same meaning just inverted.

I also like your idea about the added flags. They could then be named:
- `--disable=foo` and `--skip-publish=foo`
- or `--enable=foo` and `--publish=foo`

Finally, the commandline flags should support comma separated list and negations e.g.
- `--disable=!snapcrafts`
- `--skip-publish=dockers,brews`

What do you think?

**Comment by ivotron** [comment]
> Finally, the commandline flags should support comma separated list and negations e.g.
> 
> * `--disable=!snapcrafts`
> * `--skip-publish=dockers,brews`

heyo, sorry for resurrecting this old thread but was wondering if something like this is currently possible. thanks!

**Reply by caarlos0** [reply]
yeah, this really is something that needs to be improved.

Since this is quite a big endeavor, we'll probably need a rather big issue to map all the places we'll need to change.

If someone wants to start mapping that, we can start prioritizing it as well.
